clients/pkg/logentry/stages/pipeline.go

```diff
@@ -113,6 +114,8 @@ func RunWithSkipOrSendMany(input chan Entry, process func(e Entry) ([]Entry, boo
 // Run implements Stage
 func (p *Pipeline) Run(in chan Entry) chan Entry {
 	in = RunWith(in, func(e Entry) Entry {
+		p.mtx.Lock()
```

Comment:
Why lock `p` here?  Nothing is accessing `p` in this scope.
Maybe you meant to lock around line 127?